src/processes.c

```diff
@@ -46,76 +46,224 @@
  */
 #define G_LOG_DOMAIN "sd   main"
 
+typedef struct proc_entry
+{
+  pid_t pid;
+  int terminated;
+} proc_entry;
+
+static proc_entry *procs = NULL;
+static int initialized = 0;
+static int last;
```

Comment:
Considering that it is a relatively small array it may be better to find the last index when we need it instead of storing it in a global variable?

Reply:
Changed it